modules/tensor_mechanics/src/kernels/StressDivergencePFFracTensors.C

```diff
@@ -12,17 +12,24 @@ InputParameters validParams<StressDivergencePFFracTensors>()
 {
   InputParameters params = validParams<StressDivergenceTensors>();
   params.addClassDescription("Stress divergence kernel for phase-field fracture: Additionally computes off diagonal damage dependent Jacobian components");
-  params.addCoupledVar("c", "Phase field damage variable: Used to indicate calculation of Off Diagonal Jacobian term");
+  params.addCoupledVar("c","Phase field damage variable: Used to indicate calculation of Off Diagonal Jacobian term");
   return params;
 }
 
 
 StressDivergencePFFracTensors::StressDivergencePFFracTensors(const InputParameters & parameters) :
     DerivativeMaterialInterface<StressDivergenceTensors>(parameters),
-    _c_coupled(isCoupled("c")),
-    _c_var(_c_coupled ? coupled("c") : 0),
-    _d_stress_dc(getMaterialPropertyDerivative<RankTwoTensor>(_base_name + "stress", getVar("c", 0)->name()))
+    _c_valid(isParamValid("c")),
+    _d_stress_dc(_c_valid ? &getMaterialPropertyDerivative<RankTwoTensor>(_base_name + "stress", getVar("c", 0)->name()) : NULL)
 {
+    if (_c_valid)
+    {
+      if (isCoupled("c"))
+      {
+        _c_coupled = true;
+        _c_var = coupled("c");
```

Comment:
oh okay. I will try initializing it with false values, i.e., _c_coupled = false and _c_var = 0 and then if c is a valid parameter then I can change the values of these variables using this if condition. Thanks!